spec/features/guests/show_spec.rb

```diff
@@ -0,0 +1,151 @@
+    @guest2 = Guest.create!(name: "Zahava", nights: 3)
+    @guest3 = Guest.create!(name: "Ezzy", nights: 1)
+
+    @room1.guests << @guest1
+    @room1.guests << @guest2
+    
+    @room2.guests << @guest1
+    @room2.guests << @guest3
+
+    @room3.guests << @guest1
+  end
+
+  # =========================
+  # STORY 1: GUEST SHOW TESTS
+  # =========================
+
+  it "I see a guest's name" do
+    visit "/guests/#{@guest1.id}"
+    
+    within("div#guest_info") do
+      expect(page).to have_content(@guest1.name)
+      expect(page).to_not have_content(@guest2.name)
+    end
+    
+    visit "/guests/#{@guest2.id}"
+    
+    within("div#guest_info") do
+      expect(page).to have_content(@guest2.name)
+      expect(page).to_not have_content(@guest1.name)
+    end
+  end
+
+  it "I see a list of all the rooms they've stayed in, including information on a room's suite, nightly rate, and the name of the hotel to which it belongs" do
+    # Guest 1 Room History
+    visit "/guests/#{@guest1.id}"
+    within("div#guest_room_history") do
+      expect(page).to have_css("div#room-#{@room1.id}")
+      within("div#room-#{@room1.id}") do
+        expect(page).to have_content(@room1.suite)
+        expect(page).to have_content(@room1.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+
+      expect(page).to have_css("div#room-#{@room2.id}")
+      within("div#room-#{@room2.id}") do
+        expect(page).to have_content(@room2.suite)
+        expect(page).to have_content(@room2.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+
+      expect(page).to have_css("div#room-#{@room3.id}")
+      within("div#room-#{@room3.id}") do
+        expect(page).to have_content(@room3.suite)
+        expect(page).to have_content(@room3.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+    end
+    
+    #Guest 2 Room History
+    visit "/guests/#{@guest2.id}"
+    within("div#guest_room_history") do
+      expect(page).to have_css("div#room-#{@room1.id}")
+      within("div#room-#{@room1.id}") do
+        expect(page).to have_content(@room1.suite)
+        expect(page).to have_content(@room1.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+      
+      expect(page).to_not have_css("div#room-#{@room2.id}")
+      
+      expect(page).to_not have_css("div#room-#{@room3.id}")
+    end
+
+    # Guest 3 Room History
+    visit "/guests/#{@guest3.id}"
+    within("div#guest_room_history") do
+      expect(page).to_not have_css("div#room-#{@room1.id}")
+    
+      expect(page).to have_css("div#room-#{@room2.id}")
+      within("div#room-#{@room2.id}") do
+        expect(page).to have_content(@room2.suite)
+        expect(page).to have_content(@room2.rate)
+        expect(page).to have_content(@hotel.name)
+      end
+    
+      expect(page).to_not have_css("div#room-#{@room3.id}")
+    end
+  end
+  # =================
+  # END STORY 1 TESTS
```

Comment:
So, as thorough as these tests are, there's _a lot_ of them 😅 

I'm a fan of thoroughness, but not at the sake of time or overextending. I would recommend going through these tests again and re-assessing what, at a base level, is necessary or required for _proving_ that the user stories are implemented. Then, comment out or remove the extra parts of the tests you wrote and (even though they're good!) try to only write those necessary ones next time. We are happy to provide you feedback on that afterward if you like, just let us know.